lib/page.js

```diff
@@ -1,30 +1,34 @@
+let { repr } = require("faucet-pipeline-core/lib/util");
 let colonParse = require("metacolon");
 
-class Page {
-	constructor({ slug, file, children }, config, resolvePath) {
-		this.slug = slug;
-		this.file = resolvePath(file);
-		this.children = (children || []).map(child => {
-			child.slug = `${slug}/${child.slug}`;
-			return new Page(child, config, resolvePath);
-		});
-		this.config = config;
+module.exports = class Page {
+	constructor(slug, sourcePath, children) {
+		this.slug = slug || "";
+		this.sourcePath = sourcePath;
+		this.children = children || null;
 	}
 
-	readHeaders() {
-		return colonParse(this.file).
-			then(({ headers }) => {
-				this.description = headers.description || this.config.description;
-				this.language = headers.language || this.config.language;
-				this.heading = headers.title;
-				this.title = [headers.title, this.config.title].join(" | ");
-				return Promise.all(this.children.map(child => child.readHeaders()));
-			});
+	getChild(slug) {
+		return this.children.get(slug);
 	}
 
-	body() {
-		return colonParse(this.file).then(({ body }) => body);
+	async load() {
+		if(this.body) { // heuristic
+			return Promise.resolve(this);
+		}
+
+		let { headers, body } = await colonParse(this.sourcePath);
+		// XXX: secondary fallbacks YAGNI?
```

Comment:
So you want to allow users to either put lang or language in their frontmatter? 🤔

Reply:
Well, _I_ would only want to type `lang`, i.e. I don't want you to force me to type [_an entire word_](https://gph.is/2Mlu6Lb). Ergo, if your [cultural ignorance](https://en.wikipedia.org/wiki/MS-DOS) makes you prefer `language`, we need to support both?

Reply:
I'm pretty sure that users will rarely have files in different languages and typing `language` will be fine 😉 But we can discuss this after this is merged.